client/src/components/Common/ExportForm.vue

```diff
@@ -1,73 +1,73 @@
+<script setup lang="ts">
+import { BButton, BCol, BFormGroup, BFormInput, BRow } from "bootstrap-vue";
+import { computed, ref } from "vue";
+
+import localize from "@/utils/localization";
+
+import FilesInput from "@/components/FilesDialog/FilesInput.vue";
+
+interface Props {
+    what?: string;
+    clearInputAfterExport?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+    what: "archive",
+    clearInputAfterExport: false,
+});
+
+const emit = defineEmits<{
+    (e: "export", directory: string, name: string): void;
+}>();
+
+const directory = ref<string>("");
+const name = ref<string>("");
+
+const canExport = computed(() => name.value.length > 0 && directory.value.length > 0);
+
+const directoryDescription = computed(() => localize(`Select a 'remote files' directory to export ${props.what} to.`));
+
+const nameDescription = computed(() => localize("Give the exported file a name."));
+
+const namePlaceholder = computed(() => localize("Name"));
+
+const doExport = () => {
+    emit("export", directory.value, name.value);
+    if (props.clearInputAfterExport) {
+        directory.value = "";
+        name.value = "";
+    }
+};
+
+// This is required to make the component testable with script setup,
+// otherwise we cannot access the component's data.
+// We probably can remove this once we switch to Vue 3 or other testing framework.
+defineExpose({
+    directory,
+    name,
+    canExport,
+});
```

Comment:
It is an anti-pattern to test the components internal state. A test should only test the interface of a component, what is visible to the outside world. Sort of like a contract on what the component promises to do. How this functionality is internally implemented should be able to change without the test failing.

Here's how this might look in this case:

 - To test if the name and directory are processed correctly, look at the emitted event
 - To test if it is correctly detected if an export can be made, check if the export button is enabled

Reply:
I rewrote the tests without exposing the internals, but I still don't feel 100% clean...

In order to simulate the input changes I had to use `mount` instead of `shallowMount` (which I recall was the preferred way of testing, although that could be fine?), and also I had to explicitly import one of the child components (FilesInput) to be able to emit its custom event... maybe there is a cleaner way of simulating the input changes?